src/libsync/logger.cpp

```diff
@@ -224,14 +224,18 @@ void Logger::setLogFlush(bool flush)
 
 void Logger::setLogDebug(bool debug)
 {
-    QLoggingCategory::setFilterRules(debug ? QStringLiteral("nextcloud.*.debug=true") : QString());
+    const QSet<QString> rules = {debug ? QStringLiteral("nextcloud.*.debug=true") : QString()};
```

Comment:
@mgallien So, in case `debug` is **false**, we will end up having a `QSet` of empty `QString`, and then, few lines later, `removeLogRule` will have no effect and we will not remove `QStringLiteral("nextcloud.*.debug=true")` from the set.